src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,158 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkBase.SoftLimitDirection;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkLowLevel.PeriodicFrame;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkAbsoluteEncoder;
+import com.revrobotics.SparkAbsoluteEncoder.Type;
+import com.revrobotics.SparkPIDController;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.networktables.DoubleEntry;
+import edu.wpi.first.networktables.DoublePublisher;
+import edu.wpi.first.networktables.DoubleSubscriber;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.PubSubOption;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.lib2706.ErrorCheck;
+import frc.lib.lib2706.ProfiledPIDFFController;
+import frc.lib.lib2706.SubsystemChecker;
+import frc.lib.lib2706.SubsystemChecker.SubsystemType;
+import frc.robot.Config;
+
+public class ArmSubsystem extends SubsystemBase{
+     private static ArmSubsystem instance = null; //static object that contains all movement controls
+
+     private static final MotorType motorType = MotorType.kBrushless; //defines brushless motortype
+     public final CANSparkMax m_arm; //bottom SparkMax motor controller
+     
+     //network table entry
+     private final String m_tuningTable = "Arm/ArmTuning";
+     private final String m_dataTable = "Arm/ArmData";
+  
+     //network table entries 
+     private DoubleEntry m_armPSubs;
+     private DoubleEntry m_armISubs;
+     private DoubleEntry m_armDSubs;
+     private DoubleEntry m_armIzSubs;
+     private DoubleEntry m_armFFSubs;
+     private DoublePublisher m_armSetpointPub;   
+     private DoublePublisher m_armVelPub;
+     private DoublePublisher m_armFFTestingVolts;
+     private DoubleEntry m_armOffset;
+     private DoublePublisher m_targetAngle;
+     private DoublePublisher m_armPosPub;
+
+    //for  arm ff
+    private DoubleEntry m_armMomentToVoltage;
+
+    //spark absolute encoder
+    SparkAbsoluteEncoder m_absEncoder;  
+    //embedded relative encoder
+    //private RelativeEncoder m_Encoder;
+    public SparkPIDController m_pidControllerArm;  
+
+    ProfiledPIDFFController m_profiledFFController = new ProfiledPIDFFController();
+
+    public static ArmSubsystem getInstance() {
+      if (instance == null) {
+        SubsystemChecker.subsystemConstructed(SubsystemType.ArmSubsystem);
+        instance = new ArmSubsystem();
+      }
+      return instance;
+    }
+    public ArmSubsystem() {
```

Comment:
This constructor should be private. Making it private avoids any unnecessary mistakes with constructing multiple instances.